crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -1313,6 +1358,114 @@ pub mod tests {
+                                    "number": 1,
+                                    "hash": "hash",
+                                    "timestamp": 1
+                                }
+                            },
+                            "allocations": []
+                        }
+                    }))),
+            )
+            .await;
+
+        let server_ref = mock_escrow_subgraph_empty_response().await;
+        let (mock_escrow_subgraph_server, _mock_ecrow_subgraph) = &*server_ref;
+        let receipt =
+            create_received_receipt(&ALLOCATION_ID_0, &SIGNER.0, 1, 1, TRIGGER_VALUE - 100);
+        store_receipt(&pgpool, receipt.signed_receipt())
+            .await
+            .unwrap();
+
+        let (sender_account, notify, _, _) = create_sender_account(
+            pgpool.clone(),
+            HashSet::new(),
+            TRIGGER_VALUE,
+            TRIGGER_VALUE + 1000,
+            &mock_escrow_subgraph_server.uri(),
+            &mock_server.uri(),
+            RECEIPT_LIMIT,
+            Some(
+                Url::from_str(&("http://".to_owned() + &aggregator_endpoint.to_string()))
+                    .expect("This shouldnt fail"),
+            ),
+        )
+        .await;
+
+        // we expect it to create a sender allocation
+        sender_account
+            .cast(SenderAccountMessage::UpdateAllocationIds(
+                vec![ALLOCATION_ID_0].into_iter().collect(),
+            ))
+            .unwrap();
+        notify.notified().await;
+
+        drop(no_allocations_closed_guard);
+        mock_server
+            .register(
+                Mock::given(method("POST"))
+                    .and(body_string_contains("ClosedAllocations"))
+                    .respond_with(ResponseTemplate::new(200).set_body_json(json!({ "data": {
+                            "meta": {
+                                "block": {
+                                    "number": 1,
+                                    "hash": "hash",
+                                    "timestamp": 1
+                                }
+                            },
+                            "allocations": [
+                                {"id": *ALLOCATION_ID_0 }
+                            ]
+                        }
+                    }))),
+            )
+            .await;
+
+        // try to delete sender allocation_id
+        sender_account
+            .cast(SenderAccountMessage::UpdateAllocationIds(HashSet::new()))
+            .unwrap();
+
+        sender_account.stop_children_and_wait(None, None).await;
+
+        let rav_marked_as_last = sqlx::query!(
+            r#"
+                SELECT * FROM scalar_tap_ravs WHERE last = true;
+            "#,
+        )
+        .fetch_all(&pgpool)
+        .await
+        .expect("Should not fail to fetch from scalar_tap_ravs");
+        assert!(!rav_marked_as_last.is_empty());
+        // Stop the TAP aggregator server.
```

Comment:
This seems to be testing things from SenderAllocation. It seems to me that this is an integration test.

Reply:
Any update here?